pkg/asset/installconfig/azure/validation_test.go

```diff
@@ -162,6 +178,16 @@ func TestAzureInstallConfigValidation(t *testing.T) {
 			edits:    editFunctions{invalidateRegionLetterCase},
 			errorMsg: "region \"Central US\" is not valid or not available for this account, did you mean \"centralus\"\\?$",
 		},
+		{
+			name:     "Invalid disk types",
+			edits:    editFunctions{invalidateDiskType},
+			errorMsg: "",
```

Comment:
If there's an invalid disk type you would expect an errorMsg... so if these tests aren't failing it indicates a problem.

Reply:
These functions are not being called as they are specific to UltraSSD logic. So I left the error msg as empty.

Reply:
> Right now, this validate function is not being called. I added these test cases for future purposes.

Does the validate function work if we include it? It looked general to me (like it validates all disk types). or does it only work for Ultra SSD?

Reply:
It can work in general but usually Azure keeps all types of non-preview disks in all regions, making this check for them unnecessary(I think.. they have let me down before). I made it as general as possible for the future but right now the only use case where we would need the check would be for the UltraSSD disks which is commented.